Add notes on ROS2 approaches and add actions

BorjaOuterelo opened the pull request:
Expands Robotis projects approach regarding MW and protocol used.
Adds MW actions related to current OFERA work.

BorjaOuterelo committed:
Refs #3569. Middleware notes:
-Associates seamless integration with middleware questions.
-Adds actions regarding middleware.
-Fixes middleware and protocol names.

BorjaOuterelo committed:
Refs #3569. Extends on some ROS 2 based approaches.

BorjaOuterelo committed:
Refs #3569. Adds Renesas GR-ROSE platform.

iluetkeb commented on articles/embedded.md:
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
 | A-NUTTX-LIBCXX | Q-LANG | C++11/higher support for NuttX | Build libxx from the LLVM project on NuttX, as a pre-requisite to building rclcpp. | TODO |
 | A-PERF-RCLCPP-RESOURCE | Q-PERF | Determine resource use of rclcpp | |  TODO |
-
+| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have proved that it is a suitable protocol to seamless communicate microcontrollers with ROS 2. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

iluetkeb replied:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

iluetkeb replied:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

iluetkeb replied:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

iluetkeb replied:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

iluetkeb replied:
"proved" sounds very formal. Maybe write "demonstrate" instead. Also "seamless" -> "seamlessly".

btw, I tried adding a commit but it didn't work.

iluetkeb replied:
"proved" sounds very formal. Maybe write "demonstrate" instead. Also "seamless" -> "seamlessly".

btw, I tried adding a commit but it didn't work.

iluetkeb commented on articles/embedded.md:
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
 | A-NUTTX-LIBCXX | Q-LANG | C++11/higher support for NuttX | Build libxx from the LLVM project on NuttX, as a pre-requisite to building rclcpp. | TODO |
 | A-PERF-RCLCPP-RESOURCE | Q-PERF | Determine resource use of rclcpp | |  TODO |
-
+| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have proved that it is a suitable protocol to seamless communicate microcontrollers with ROS 2. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

iluetkeb replied:
I found "proved", with its connotation of formality a bit too strong. "demonstrate" should suffice.

iluetkeb commented on articles/embedded.md:
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
 
 ### ROS2-based approaches
-*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the micro-XRCE-DDS implementation of the upcoming XRCE-DDS standard. In ...
+*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In ...
```
Have you guys reached a consensus on the use of the domains with Erle? Personally, I would prefer linking to the Github-page for micro-ROS only.

BorjaOuterelo replied:
Yes, we agreed to use Github provided domain https://microros.github.io/micro-ROS/

iluetkeb commented on articles/embedded.md:
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
 
 ### ROS2-based approaches
-*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the micro-XRCE-DDS implementation of the upcoming XRCE-DDS standard. In ...
+*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In ...
```
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [https://microros.github.io/micro-ROS/](https://microros.github.io/micro-ROS/).
```
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [http://micro-ros.com](http://micro-ros.com).
```

iluetkeb replied:
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [https://microros.github.io/micro-ROS/](https://microros.github.io/micro-ROS/).
```

iluetkeb replied:
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [https://microros.github.io/micro-ROS/](https://microros.github.io/micro-ROS/).
```

iluetkeb commented on articles/embedded.md:
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
 
-*   [ROS 2 library for OpenCR by ROBOTIS](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ROS2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2)
+*   [ROS 2 library for OpenCR by ROBOTIS](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2):
+  Project enabling communication between Robotis' OpenCR board and ROS 2 nodes as part of the TurtleBot3 platform. This communication is enabled using DDS-XRCE implementation, Micro XRCE-DDS.
```
```suggestion
  Project enabling communication between Robotis' OpenCR board and ROS 2 nodes as part of the TurtleBot3 platform. This communication is enabled using a DDS-XRCE implementation, Micro XRCE-DDS.
```

iluetkeb reviewed (changes requested):
I like the additions. A few grammatical changes are suggested.

One note that I would like a comment on is the use of the micro-ros.com domain. I thought we had decided to use the Github IO page only. I may be wrong, of course, please clarify.

iluetkeb committed:
Update articles/embedded.md

Co-Authored-By: BorjaOuterelo <borjaouterelo@gmail.com>

BorjaOuterelo committed:
Refs #3569. Resolves pull requests comments:
-Fixes some language misuse.
-Splits sentences to single lines to improve diff readability.

ralph-lange commented on articles/embedded.md:
```diff
@@ -67,7 +67,8 @@ To answer these questions, the OFERA EU project as well as several others have a
 | A-NUTTX-LIBCXX | Q-LANG | C++11/higher support for NuttX | Build libxx from the LLVM project on NuttX, as a pre-requisite to building rclcpp. |
 | A-PERF-RCLCPP-RESOURCE | Q-PERF | Determine resource use of rclcpp | |
-
+| A-DDS-XRCE | Q-COMM | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```
Is the order DDS XRCE correct? I.e. the standard is named DDS XRCE whereas your implementation is named Micro XRCE-DDS?

BorjaOuterelo replied:
Hi @ralph-lange. Yes, that's correct. OMG standard acronym is DDS-XRCE, and eProsima's product name is Micro XRCE-DDS.

ralph-lange commented on articles/embedded.md:
```diff
@@ -76,10 +77,21 @@ To answer these questions, the OFERA EU project as well as several others have a
 
 *   [**ros2_embedded_nuttx**](https://github.com/ros2/ros2_embedded_nuttx): Early port (in 2014) of ROS 2 alpha for the STM32F4Discovery board and the STM3240G eval board running the RTOS NuttX developed by Víctor Mayoral Vilches and Esteve Fernandez at the OSRF.
 
+*   [Renesas GR-ROSE](http://gadget.renesas.com/ja/event/2018/pm_rose.html):
```
Renesas GR-ROSE in bold, please.

BorjaOuterelo replied:
Done

ralph-lange commented on articles/embedded.md:
```diff
@@ -76,10 +77,21 @@ To answer these questions, the OFERA EU project as well as several others have a
 
 *   [**ROS 2 library for OpenCR by ROBOTIS**](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2): Tailored and optimized implementation of the ROS 2 publish/subscribe and clock API for the Micro XRCE-DDS (formerly micro-RTPS) implementation of the upcoming XRCE-DDS middleware standard running on an STM32F7.
 
+*   [XEL Network by ROBOTIS](https://xelnetwork.readthedocs.io):
```
XEL Network by Robotis in bold, please, for consistency.

BorjaOuterelo replied:
Done

BorjaOuterelo commented:
@iluetkeb I have addressed your observations.

BorjaOuterelo committed:
Refs #3569. Makes the text consistent.

BorjaOuterelo commented:
@iluetkeb @ralph-lange I pushed latest changes following your recommendations. Can you check and we merge this before the ROS 2 PR?.
